src/com/zetcode/Board.java

```diff
@@ -212,6 +219,8 @@ private void checkCollision() {
                 ball.setXDir(1);
                 ball.setYDir(-1);
             }
+
+            Ball.playSound();
```

Comment:
With the isolation of the SoundPlayer you would then just add the call to PlaySound here.